derive_builder_core/src/builder.rs

```diff
@@ -136,12 +168,47 @@ impl<'a> ToTokens for Builder<'a> {
             #[cfg(not(feature = "clippy"))]
             tokens.append_all(quote!(#[allow(clippy::all)]));
 
+            let builder_error_ident = format_ident!("{}Error", builder_ident);
+            let builder_error_doc = format!("Error type for {}", builder_ident);
+
             tokens.append_all(quote!(
                 #[derive(#derived_traits)]
                 #builder_doc_comment
                 #builder_vis struct #builder_ident #struct_generics #where_clause {
                     #(#builder_fields)*
                 }
+
+                #[doc=#builder_error_doc]
+                #[derive(Debug)]
```

Comment:
Does it make sense for this to also be marked `#[non_exhaustive]`? I know that would be an issue for older Rust versions, but not doing so would mean that changes to the enum in future would cause downstream crates to have to increment their versions.

Reply:
Done